Simplify setup.py and move towards cookie-cutter configuration

agoose77 opened the pull request:
This PR bumps the dependency versions of the JS package to follow the released versions rather than RCs.
It also further simplifies setup.py to follow the structure of the cookie-cutter used for prebuilt extensions.

agoose77 committed:
Feature: bump dependencies

agoose77 committed:
Refactor: change labextension output dir, and simplify setup.py

agoose77 committed:
Refactor: use setuptools.find_packages()

maartenbreddels commented on setup.py:
```diff
@@ -11,45 +8,29 @@
-        ('share/jupyter/nbextensions/jupyter-threejs',
-         name + '/static',
-         '*.js.map'),
-        ('share/jupyter/lab/extensions',
-         'js/lab-dist',
-         'jupyter-threejs-*.tgz'),
```
i guess we still want lab2 support?

agoose77 replied:
Yes, I suppose we do (please excuse the slash-and-burn approach :P)

maartenbreddels commented on js/package.json:
```diff
@@ -54,7 +54,7 @@
   "jupyterlab": {
     "extension": "src/jupyterlab-plugin",
-    "outputDir": "../share/jupyter/labextensions/jupyter-threejs",
+    "outputDir": "../pythreejs/labextension",
```
This doesn't match the `"share/jupyter/labextensions/jupyter-threejs"` in setup.py right?

agoose77 replied:
Ah, looks like some of my changes have mixed up... I'll resolve this

agoose77 replied:
I think this is correct: the outputdir maps to `<ROOT>/pythreejs/labextension`, and `setup.py` looks for `<ROOT>/pythreejs/labextension`

maartenbreddels reviewed (commented):
Thnx for picking this up!

agoose77 committed:
Bugfix: restore JupyterLab2 extensions

agoose77 commented:
@maartenbreddels I've restored support for JupyterLab2 extensions, and replied to your query about the extension; both the setup.py and package.json refer to the same (<ROOT>/pythreejs/labextension) directory

maartenbreddels commented on setup.py:
```diff
@@ -11,45 +8,31 @@
-        ('share/jupyter/labextensions/jupyter-threejs/static',
-         'share/jupyter/labextensions/jupyter-threejs/static/',
-         '*.*'),
-        ('etc/jupyter/nbconfig',
-         'jupyter-config',
-         '**/*.json'),
```
why is this removed, i guess we still want that right?

bollwyvl replied:
yeah, should keep.

bollwyvl replied:
I put this under test on the [github actions PR](https://github.com/jupyter-widgets/pythreejs/pull/349/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR171)

agoose77 replied:
Yes, this was an accidental remove :/ Will fix.

agoose77 replied:
Resolved by https://github.com/maartenbreddels/pythreejs/pull/2/commits/8b0a734e78f0f1568b23904d8e5de0d37d73d71a

maartenbreddels commented on js/package.json:
```diff
@@ -54,7 +54,7 @@
   "jupyterlab": {
     "extension": "src/jupyterlab-plugin",
-    "outputDir": "../share/jupyter/labextensions/jupyter-threejs",
+    "outputDir": "../pythreejs/labextension",
```
I don't think this makes things simpler to understand, having the same directory keeps things simple.

agoose77 replied:
OK, will revert this :)

bollwyvl replied:
the only concern might be path length, if it's getting deep, for windows folks. 256 chars comes up so fast... and frigging CI machines already have it set, so it's up to users to discover...

agoose77 replied:
I'm not a windows user these days, but I can appreciate the concern; @maartenbreddels do you feel like comfortable leaving it in-tree?

maartenbreddels replied:
ok, lets leave it as is.

bollwyvl commented on js/package.json:
```diff
@@ -34,14 +34,14 @@
   },
   "devDependencies": {
-    "@jupyterlab/builder": "^3.0.0-rc.13",
+    "@jupyterlab/builder": "^3.0.0",
```
needs bump to at least `3.0.1`

agoose77 replied:
@bollwyvl what's the issue with the 3.0.0 release?

bollwyvl replied:
The last time i was working down at this level, it complained to me when building... but i can't find where it applies the check in core...

bollwyvl commented on setup.py:
```diff
@@ -91,7 +74,7 @@
         ]
     },
-    'packages': [name],  # Manually specify here, update after autogen
+    'packages': setuptools.find_packages(),
```
this is always a bit fraught... it would be good to check exactly what goes in, and potentially add some `exclude`. Moving to a `src/` layout helps this _a lot_, but is probably out of scope for this.

agoose77 replied:
Yeah, let's push any major layout changes that a different PR :)

vidartf replied:
Was this motivated by something? I think the comment is still important, as *the package list made here won't include the autogenerated python subpackages* no matter which method we choose. I see the code of the pythreejs PR simply removed this important step: https://github.com/jupyter-widgets/pythreejs/pull/342#discussion_r575259828

agoose77 replied:
I seem to have dropped the ball on this; I tested things locally but something must have been cached. I'll revert the change.

bollwyvl commented:
i still don't know what it does for us, but `install.json` seems to be showing up in other places. more metadata good!

agoose77 commented:
I'll address the comments tomorrow GMT :)

agoose77 committed:
Bugfix: restore Jupyter Notebook support

Also, convert pathlib.Path instances to str for consistency

agoose77 committed:
Revert use of find_packages

agoose77 committed:
WIP: monkeypatch jupyter_packaging to use our modified _wrap_command

This might not be the preferred solution, but I thought it was better
to have a lightweight monkeypatch than to reimplement half of setupbase

agoose77 committed:
Bugfix: remove deleted file from MANIFEST.in

maartenbreddels commented:
why is https://github.com/maartenbreddels/pythreejs/pull/2/commits/5eee7d1651d4178b5e5e271f646c5bc7fbbc65f1 needed? I'm afraid if we don't keep this PR minimal, we will never ship this.

agoose77 commented:
@maartenbreddels it's a fair concern!

I made an mistake when removing the bundled `setupbase.py` in favour of the `jupyter_packaging` one. I'm not really sure what happened because I tested everything locally, but I must have missed something. The last few commits https://github.com/maartenbreddels/pythreejs/pull/2/commits/8b0a734e78f0f1568b23904d8e5de0d37d73d71a ... https://github.com/maartenbreddels/pythreejs/pull/2/commits/adc106ec87666cabda8799a2790d8684ef8ab379 are all to fix these oversights. I kept them in separate commits to keep the history clean.

In particular, https://github.com/maartenbreddels/pythreejs/commit/5eee7d1651d4178b5e5e271f646c5bc7fbbc65f1 just restores the `pythreejs` specific change to `setupbase.py` by monkeypatching it. Unfortunately there isn't a cleaner way to do this as far as I can see, and I think it's important to avoid having a separate implementation of `jupyter_packaging`

The [files view](https://github.com/maartenbreddels/pythreejs/pull/2/files) gives a better overview of how simple this PR is! :smile:

maartenbreddels commented:
Can you revert that removal?

(from mobile phone)

On Fri, Feb 12, 2021, 21:31 Angus Hollands <notifications@github.com> wrote:

> @maartenbreddels <https://github.com/maartenbreddels> it's a fair concern!
>
> I made an mistake when removing the bundled setupbase.py in favour of the
> jupyter_packaging one. I'm not really sure what happened because I tested
> everything locally, but I must have missed something. The last few commits
> 8b0a734
> <https://github.com/maartenbreddels/pythreejs/commit/8b0a734e78f0f1568b23904d8e5de0d37d73d71a>
> ... adc106e
> <https://github.com/maartenbreddels/pythreejs/commit/adc106ec87666cabda8799a2790d8684ef8ab379>
> are all to fix these oversights. I kept them in separate commits to keep
> the history clean.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/maartenbreddels/pythreejs/pull/2#issuecomment-778438308>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANPEPL4Z6HMXNSEOBSBHP3S6WFZJANCNFSM4WRP5W5Q>
> .
>

agoose77 commented:
I can, would you rather bundle an entire `setupbase.py` vs depending upon `jupyter_packaging`?

maartenbreddels commented:
I'd rather see that go in a different PR if it gives trouble, so we can
focus on jlab3 in this pr.

(from mobile phone)

On Fri, Feb 12, 2021, 22:01 Angus Hollands <notifications@github.com> wrote:

> I can, would you rather bundle an entire setupbase.py vs depending upon
> jupyter_packaging?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/maartenbreddels/pythreejs/pull/2#issuecomment-778452573>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANPEPPTVQCH4QURL36DHIDS6WJJHANCNFSM4WRP5W5Q>
> .
>

agoose77 commented:
OK, thanks for all the input everyone. I'm going to close this, because the revert would need to happen in your branch (https://github.com/jupyter-widgets/pythreejs/pull/342/commits/df6afcfa0e82f3cd9d996526f672b6e48e8f6298)... and the history is getting messy! I will open a new PR on jupyter-widgets/pythreejs  that builds on top of 49ffb97
